### DISCUSSION TITLE
Migration from Eclipse GLSP from 1.1.0.RC10 into 2.2.1 (EMFModelState with getInjector)

### DISCUSSION THREAD

**Discussion by microwan** [eclipse-glsp/glsp#1424] [General] [answered]
Hello ,

we are trying to migrate our tools from Eclipse GLSP 1.1.0.RC09 into 2.2.1

We have a lot of code getInjector()
How to retrieve the EMFModelState

```private PinCfgGLSPDiagramEditor editor;

EMFModelState state = editor.getInjector().getInstance(EMFModelState.class);

how to retrieve easily the EMFModelState of our custom Diagram Editor
```
 Best regards

 Erwan

**Comment by tortmayr** [comment, marked as answer]
Hi @microwan,

this has changed with the introduction of the [E4 support](https://github.com/eclipse-glsp/glsp-eclipse-integration/pull/83)
A lot functionality has been moved from the editor into the `GLSPDiagramComposite` composite.

To retrieve the model state via  injector from the editor you need to do it this way now:

```java
editor.getDiagram().getInjector().getInstance(EMFModelState.class);
```
